examples/App.re

```diff
@@ -9,69 +7,79 @@ type state = {route};
 type action =
   | UpdateRoute(route);
 
-let getInitialRoute = () =>
-  switch (locationHash) {
-  | "#signup" => Signup
-  | "#login" => Login
-  | _ => Signup
-  };
+[@react.component]
+let make = () => {
+  let initialRoute =
+    React.useMemo1(
+      () => {
+        let url = ReasonReactRouter.dangerouslyGetInitialUrl();
+        switch (url.hash) {
+        | "signup" => Signup
+        | "login" => Login
+        | _ => Signup
+        };
+      },
+      [||],
+    );
 
-let component = React.reducerComponent(__MODULE__);
+  let (state, dispatch) =
+    React.useReducer(
+      (_, action) =>
+        switch (action) {
+        | UpdateRoute(nextRoute) => {route: nextRoute}
+        },
+      {route: initialRoute},
+    );
 
-let make = _ => {
-  ...component,
-  initialState: () => {route: getInitialRoute()},
-  reducer: (action, _) =>
-    switch (action) {
-    | UpdateRoute(route) => React.Update({route: route})
+  React.useEffect1(
```

Comment:
you could use `React.useEffect0` here (https://github.com/reasonml/reason-react/blob/bb480418d99acc49bb5df5df512f68646cce26ad/src/React.re#L131) this sends an empty array as argument.

Reply:
TIL, will do 👍